playbooks/provisioning/aws/cloudformation/files/nested/control-plane/default.yaml

```diff
@@ -0,0 +1,229 @@
+        Ebs:
+          DeleteOnTermination: True
+          VolumeSize:
+            Ref: MasterEtcdVolSize
+          VolumeType: gp2
+      UserData:
+        Ref: MasterUserData
+  Master02:
+    Type: 'AWS::EC2::Instance'
+    Properties:
+      ImageId:
+        Ref: MasterImageId
+      KeyName:
+        Ref: KeyName
+      InstanceType:
+        Ref: MasterInstanceType
+      SecurityGroupIds:
+        Ref: MasterSGs
+      SubnetId:
+        Ref: Master02Subnet
+      IamInstanceProfile:
+        Ref: MasterInstanceProfile
+      BlockDeviceMappings:
+      - DeviceName: /dev/sda1
+        Ebs:
+          DeleteOnTermination: True
+          VolumeSize:
+            Ref: MasterRootVolSize
+          VolumeType: gp2
+      - DeviceName: /dev/xvdb
+        Ebs:
+          DeleteOnTermination: True
+          VolumeSize:
+            Ref: MasterDockerVolSize
+          VolumeType: gp2
+      - DeviceName: /dev/xvdc
+        Ebs:
+          DeleteOnTermination: True
+          VolumeSize:
+            Ref: MasterEtcdVolSize
+          VolumeType: gp2
+      UserData:
+        Ref: MasterUserData
+  Master03:
+    Type: 'AWS::EC2::Instance'
+    Properties:
+      ImageId:
+        Ref: MasterImageId
+      KeyName:
+        Ref: KeyName
+      InstanceType:
+        Ref: MasterInstanceType
+      SecurityGroupIds:
+        Ref: MasterSGs
+      SubnetId:
+        Ref: Master03Subnet
+      IamInstanceProfile:
+        Ref: MasterInstanceProfile
+      BlockDeviceMappings:
+      - DeviceName: /dev/sda1
+        Ebs:
+          DeleteOnTermination: True
+          VolumeSize:
+            Ref: MasterRootVolSize
+          VolumeType: gp2
+      - DeviceName: /dev/xvdb
+        Ebs:
+          DeleteOnTermination: True
+          VolumeSize:
+            Ref: MasterDockerVolSize
+          VolumeType: gp2
+      - DeviceName: /dev/xvdc
+        Ebs:
+          DeleteOnTermination: True
+          VolumeSize:
+            Ref: MasterEtcdVolSize
+          VolumeType: gp2
+      UserData:
+        Ref: MasterUserData
+  MasterIntElb:
+    Type: 'AWS::ElasticLoadBalancing::LoadBalancer'
+    Properties:
+      CrossZone: True
+      ConnectionSettings:
+        IdleTimeout: 3600
+      Listeners:
+      - InstancePort:
+          Ref: MasterApiPort
+        InstanceProtocol: TCP
+        LoadBalancerPort:
+          Ref: MasterApiPort
+        Protocol: TCP
+      Scheme: internal
+      SecurityGroups:
+        Ref: MasterIntElbSGs
+      Subnets:
+      - Ref: Master01Subnet
+      - Ref: Master02Subnet
+      - Ref: Master03Subnet
+      Instances:
+      - Ref: Master01
+      - Ref: Master02
+      - Ref: Master03
+      HealthCheck:
+        HealthyThreshold: 2
+        Interval: 5
+        Timeout: 2
+        UnhealthyThreshold: 2
+        Target:
+          'Fn::Join':
+          - ''
+          - - 'Fn::Join':
+              - ':'
+              - - 'HTTPS'
+                - Ref: MasterApiPort
+            - /healthz/ready
```

Comment:
No, no need to specify the hostname, This specifies an https health check using port 443 and the path /healthz/ready. While it looks odd, it is the correct format for an ELB health check target.